docs/middleware.md

```diff
@@ -264,6 +264,7 @@ around explicitly, rather than mutating the middleware instance.
 Currently, the `BaseHTTPMiddleware` has some known limitations:
 
 - Using `BaseHTTPMiddleware` will prevent changes to [`contextlib.ContextVar`](https://docs.python.org/3/library/contextvars.html#contextvars.ContextVar)s from propagating upwards. That is, if you set a value for a `ContextVar` in your endpoint and try to read it from a middleware you will find that the value is not the same value you set in your endpoint (see [this test](https://github.com/encode/starlette/blob/621abc747a6604825190b93467918a0ec6456a24/tests/middleware/test_base.py#L192-L223) for an example of this behavior).
+- Using `BaseHTTPMiddleware` will prevent [ASGI pathsend extension](https://asgi.readthedocs.io/en/latest/extensions.html#path-send) to work properly. Thus, if you run your Starlette application with a server implementing this extension, routes returning [FileResponse](responses.md#fileresponse) should avoid the usage of this middleware.
```

Comment:
FWIW I think https://github.com/encode/starlette/pull/2616/files#r1684224798 would be made easier / resolved by https://github.com/encode/starlette/pull/2620 because then we can change it only for `BaseHTTPMiddleware`

Reply:
@adriangb I think it still really depends on wether `BaseHTTPMiddleware` only supports `response.body` messages or not..

Reply:
@gi0baro I think you can change the `BaseHTTPMiddleware` now, can't you?

Reply:
Yep my point was that since it how has it's own Response class it should be easier to edit :)

Reply:
👍 gonna rebase and edit `BaseHTTPMiddleware`